Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# No Randomly Generated Families for Lore-Important Characters

## Changelog:
- Added exceptions to certain lore-important characters to no longer generate random families at the start of a game.

**Full list of included characters**:

<details><summary>Click to expand</summary>

                    ## Humans
			AND = {
				exists = character:60036
				this = character:60036 # Uther the Lightbringer
			}
			AND = {
				exists = character:48435
				this = character:48435 # Priscilla Ashvane
			}
			AND = {
				exists = character:59000
				this = character:59000 # Archmage Antonidas
			}
			AND = {
				exists = character:5000
				this = character:5000 # Medivh
			}
			AND = {
				exists = character:60014
				this = character:60014 # Tirion Fordring
			}
			AND = {
				exists = character:38002
				this = character:38002 # Galen Trollbane
			}
			AND = {
				exists = character:46100
				this = character:46100 # Arugal
			}
			
			
			## Orcs & Ogres
			
			AND = {
				exists = character:10021
				this = character:10021 # Go'el/Thrall
			}
			AND = {
				exists = character:10005
				this = character:10005 # Kilrogg
			}
			AND = {
				exists = character:52000
				this = character:52000 # Cho'gall
			}
			AND = {
				exists = character:10015
				this = character:10015 # Gul'dan
			}
			AND = {
				exists = character:10050
				this = character:10050 # Orgrim
			}
			AND = {
				exists = character:10002
				this = character:10002 # Dal'rend Blackhand
			}
			AND = {
				exists = character:10100
				this = character:10100 # Drek'thar
			}
			AND = {
				exists = character:10656
				this = character:10656 # Rehgar Earthfury
			}
			
			
			## Dwarves
			AND = {
				exists = character:12002
				this = character:12002 # Dagran Thaurissan
			}
			AND = {
				exists = character:22001
				this = character:22001 # Falstad Wildhammer
			}
			
			
			## Night Elves
			AND = {
				exists = character:340018
				this = character:340018 # Fandral Staghelm
			}
			AND = {
				exists = character:340003
				this = character:340003 # Tyrande (Theocracy)
			}
			AND = {
				exists = character:340100
				this = character:340100 # Shandris Feathermoon
			}
			
			
			## High Elves/Blood Elves
			AND = {
				exists = character:42077
				this = character:42077 # Halduron Brightwing (minor-ish character, maybe generated family is okay)
			}
			AND = {
				exists = character:42078
				this = character:42078 # Dar'Khan Drathir (minor-ish character, maybe generated family is okay)
			}
			AND = {
				exists = character:42304
				this = character:42304 # Grand Magister Belo'vir
			}
			AND = {
				exists = character:42075
				this = character:42075 # Lor'themar Theron
			}
			AND = {
				exists = character:42306
				this = character:42306 # Grand Magister Rommath
			}
			AND = {
				exists = character:42003
				this = character:42003 # Kael'thas Sunstrider
			}
			
			
			## Dragons/Draconic
			AND = {
				exists = character:62202
				this = character:62202 # Nefarian (Theocracy)
			}
			AND = {
				exists = character:62203
				this = character:62203 # Onyxia
			}
			AND = {
				exists = character:62400
				this = character:62400 # Alexstrasza (Theocracy)
			}
			AND = {
				exists = character:62401
				this = character:62401 # Korialstrasz (Theocracy)
			}
			
			
			## Trolls
			AND = {
				exists = character:57000
				this = character:57000 # Zul'jin
			}
			AND = {
				exists = character:68001
				this = character:68001 # Vol'jin
			}
			
			## Demons
			AND = {
				exists = character:430400
				this = character:430400 # Balnazzar
			}
			AND = {
				exists = character:360000
				this = character:360000 # Kazzak
			}
			
			## Undead
			AND = {
				exists = character:5806
				this = character:5806 # Moroes
			}
			AND = {
				exists = character:10014
				this = character:10014 # Ner'zhul (the Lich, not the Orc)
			}

</details>

## What does this mean?

Normally, **ruler characters** without any pre-existing children will have random families generated for them when the game starts. Think someone like Medivh or Uther. We don't know of them having any children, they appear as rulers in some bookmarks, and so the game would normally spawn them in with a spouse and some children.

Adding them to this list ensures they will not have spouses or children generated in any bookmark in which they appear as rulers. Additionally, Scourge characters will not generate families by default either, so they do not need to be added (unless they appear as non-Scourge rulers in earlier bookmarks).


# How to test:
- Make suggestions on characters that should be either added to or removed from this list of exceptions.

To look for applicable characters, start the game with [Debug Mode](https://ck3.paradoxwikis.com/Console_commands#Debug_mode) on and use the Character Finder to find lore-important rulers with children that **do not** have a Historical ID (this means they were generated by the game).

**Note**: We do not want to add every lore character to this list of exceptions. With most minor lore characters and NPCs, we **_want_** them to have randomly generated families.
